src/main.rs

```diff
@@ -1,40 +1,43 @@
 use anyhow::Result;
-use clap::Parser;
-use fuelup::commands::install;
+use fuelup::download::component;
+use fuelup::{fuelup_cli, proxy_cli};
+use std::panic;
+use std::path::PathBuf;
 
-use fuelup::commands::fuelup::{self_update, FuelupCommand};
-use fuelup::commands::install::InstallCommand;
-
-#[derive(Debug, Parser)]
-#[clap(name = "fuelup", about = "Fuel Toolchain Manager", version)]
-struct Cli {
-    #[clap(subcommand)]
-    command: Commands,
+fn is_supported_component(component: &str) -> bool {
+    ["forc", "fuel-core", "forc-fmt", "forc-lsp", "forc-explore"].contains(&component)
 }
 
-#[derive(Debug, Parser)]
-enum Commands {
-    /// Installs the latest Fuel toolchain.
-    Install(InstallCommand),
-    /// Manage your fuelup installation.
-    #[clap(name = "self", subcommand)]
-    Fuelup(FuelupCommand),
+fn run() -> Result<()> {
+    let arg0 = std::env::args().next().map(PathBuf::from);
+
+    let process_name = arg0
+        .as_ref()
+        .and_then(|a| a.file_stem())
+        .and_then(std::ffi::OsStr::to_str)
+        .map(String::from);
+
+    match process_name.as_deref() {
+        Some(component::FUELUP) => fuelup_cli::fuelup_cli()?,
+        Some(n) => {
+            if is_supported_component(n) {
+                proxy_cli::proxy_run(n)?;
+            }
+        }
+        None => panic!("fuelup does not understand this command"),
+    }
+    Ok(())
 }
 
-fn main() -> Result<()> {
+fn main() {
     let format = tracing_subscriber::fmt::format()
         .without_time()
         .with_level(false)
         .with_target(false);
 
     tracing_subscriber::fmt().event_format(format).init();
 
-    let cli = Cli::parse();
-
-    match cli.command {
-        Commands::Install(_command) => install::install(),
-        Commands::Fuelup(command) => match command {
-            FuelupCommand::Update => self_update(),
-        },
+    if run().is_err() {
```

Comment:
Hmm, I left this empty because the thought was the error messages propagated from the inner process might've be enough - it might be annoying if fuelup prints an extra error message along with the original message from the inner process?

Reply:
Perhaps this should be handled within the `run()`? Maybe an error message in the match arm of `fuelup_cli::fuelup_cli()`

Reply:
Ahh I see what you mean! Yes it would be nice to `error!` in the case that `fuelup` itself was run, but in the case that it was used as a proxy, we allow the child process to print errors as necessary. Note that we do something similar for `forc` plugins, where `forc` similarly acts as a proxy to some child process.